registrations/models.py

```diff
@@ -250,7 +250,40 @@ def is_user_editable_resources(self):
         )
 
 
-class SignUp(CreatedModifiedBaseModel):
+class SignUpMixin:
+    @property
+    def data_source(self):
+        return self.registration.data_source
+
+    @property
+    def publisher(self):
+        return self.registration.publisher
+
+    def can_be_edited_by(self, user):
+        """Check if the current signup can be edited by the given user"""
+        return (
+            user.is_superuser
+            or user.is_admin_of(self.publisher)
+            or user.id == self.created_by_id
+        )
+
+    def is_user_editable_resources(self):
+        return bool(self.data_source and self.data_source.user_editable_resources)
+
+
+class SignUpGroup(CreatedModifiedBaseModel, SignUpMixin):
+    registration = models.ForeignKey(
+        Registration, on_delete=models.PROTECT, related_name="signup_groups"
+    )
+    extra_info = models.TextField(
+        verbose_name=_("Extra info"),
+        blank=True,
+        null=True,
```

Comment:
Sonarcloud complains about having a nullable text field, but it's pretty much in line with the existing code base so I don't see a need to change it. If no changes will be made then feel free to close the sonarcloud issue.